fsmlib.h

```diff
@@ -85,31 +103,46 @@ name##_fsm_t name##_fsm[] = {                               \
     __VA_ARGS__                                             \
 };                                                          \
                                                             \
-static name##_states_t        state = 0;                    \
+static name##_states_t        name##_state = 0;             \
 static name##_fsm_t         * name##_transition;            \
                                                             \
+name##_states_t name##_fsm_get_state(){                     \
+    return name##_state;                                    \
+}                                                           \
+                                                            \
+int8_t name##_fsm_set_state(name##_states_t new_state){     \
+    if ((new_state >= 0) & (new_state < name##_STATE_COUNT)) { \
+        name##_state = new_state;                           \
+        return FSM_OK;                                      \
+    }                                                       \
+    return FSM_INVALID_STATE;                               \
+}                                                           \
```

Comment:
States should only be changeable by transitions (events) through the private step function. A public function would bypass the logic of fsm.